File: src/com/timepath/quakec/psi/QCTokenType.java
package com.timepath.quakec.psi;

import com.intellij.psi.tree.IElementType;
import com.timepath.quakec.QCLanguage;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

/**
 * @author TimePath
 */
public class QCTokenType extends IElementType {
    public QCTokenType(@NotNull @NonNls String debugName) {
        super(debugName, QCLanguage.INSTANCE);
    }

    @Override
    public String toString() {
        return "QCTokenType." + super.toString();
    }
}

File: src/main/kotlin/com/timepath/quakec/psi/QCTokenType.kt
package com.timepath.quakec.psi

import com.intellij.psi.tree.IElementType
import com.timepath.quakec.ide.QCLanguage
import org.jetbrains.annotations.NonNls

class QCTokenType(@NonNls debugName: String) : IElementType(debugName, QCLanguage) {

    override fun toString() = "TokenTypeImpl.${super.toString()}"
}
